### ISSUE TITLE
Building with Node 18 

### ISSUE THREAD

**Issue by bergmorten** [aws-amplify/amplify-hosting#3796]
### How did you install the Amplify CLI?

npm

### If applicable, what version of Node.js are you using?

node 18.x 

### Amplify CLI Version

12.7.1

### What operating system are you using?

windows

### Did you make any manual changes to the cloud resources managed by Amplify? Please describe the changes made.

No

### Describe the bug

I've build setting with latest amplify CLI and node 18, but this will not build when patching/setting amplify version...

<img width="1035" alt="image" src="https://github.com/aws-amplify/amplify-cli/assets/86950981/e1badd25-e18e-463c-8c04-dffe3372831c">

```
2023-11-09T13:50:45.750Z [INFO]: SSM params {"Path":"/amplify/d3535s7bikhhjd/develop/","WithDecryption":true}
2023-11-09T13:51:09.159Z [INFO]: # Node version 18 is available for installation
2023-11-09T13:51:09.324Z [INFO]: # Installing Node version 18
2023-11-09T13:51:14.031Z [INFO]: # Now using Node version 18
2023-11-09T13:51:15.858Z [WARNING]: Cannot find module '/root/.nvm/versions/node//lib/node_modules/@aws-amplify/cli/package.json'
                                    Require stack:
                                    - /snapshot/AWSMobilePioneerExecute/out/index.js
2023-11-09T13:51:15.859Z [INFO]: # Patching NPM package '@aws-amplify/cli' to 12.8.2...
2023-11-09T13:51:16.418Z [INFO]: node: /lib64/libm.so.6: version `GLIBC_2.27' not found (required by node)
                                 node: /lib64/libc.so.6: version `GLIBC_2.28' not found (required by node)
2023-11-09T13:51:16.418Z [ERROR]: !!! Error patching NPM package '@aws-amplify/cli' to version latest. Verify the package name and version are correct, and retry the build. For help see our docs: https://docs.aws.amazon.com/amplify/latest/userguide/custom-build-image.html
```

Changing back to Node 16, and then i works. Since newest LTS is version 20,  node 18 should at least be supported?

### Expected behavior

Should build when build setting is node 18 and amplify cli set to latest

### Reproduction steps

Change build setting to node 18 and amplify to latest, then to new build or redeploy.

<img width="1035" alt="image" src="https://github.com/aws-amplify/amplify-cli/assets/86950981/7dc5a798-5010-4d61-95f4-29d9f16fde8a">

### Log output

<details>

```
# Put your logs below this line
2023-11-09T13:50:45.750Z [INFO]: SSM params {"Path":"/amplify/d3535s7bikhhjd/develop/","WithDecryption":true}
2023-11-09T13:51:09.159Z [INFO]: # Node version 18 is available for installation
2023-11-09T13:51:09.324Z [INFO]: # Installing Node version 18
2023-11-09T13:51:14.031Z [INFO]: # Now using Node version 18
2023-11-09T13:51:15.858Z [WARNING]: Cannot find module '/root/.nvm/versions/node//lib/node_modules/@aws-amplify/cli/package.json'
                                    Require stack:
                                    - /snapshot/AWSMobilePioneerExecute/out/index.js
2023-11-09T13:51:15.859Z [INFO]: # Patching NPM package '@aws-amplify/cli' to 12.8.2...
2023-11-09T13:51:16.418Z [INFO]: node: /lib64/libm.so.6: version `GLIBC_2.27' not found (required by node)
                                 node: /lib64/libc.so.6: version `GLIBC_2.28' not found (required by node)
2023-11-09T13:51:16.418Z [ERROR]: !!! Error patching NPM package '@aws-amplify/cli' to version latest. Verify the package name and version are correct, and retry the build. For help see our docs: https://docs.aws.amazon.com/amplify/latest/userguide/custom-build-image.html

```

</details>

### Before submitting, please confirm:

- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.
- [X] I have removed any sensitive information from my code snippets and submission.

**Comment by josdejong** [comment]
I found a solution that works for my case (using a frontend only):

Go to Build settings -> Build image settings -> Edit. Select "Build image" instead of "Amaon Linux:2 (default)", enter "public.ecr.aws/docker/library/node:20.9.0", and clear the Live package updates (In my case there was one for "Nodejs version 16"):

![afbeelding](https://github.com/aws-amplify/amplify-hosting/assets/568626/f76bfae7-3cea-4ba3-b95f-d9c022fbcc85)

**Comment by bergmorten** [comment]
That would work :-)

I still think Amplify should change default image to Linux 2023 (which I think has support for both node 18 and 20). It seems that node 18 is the default node version inside amplify when creating lambda functions, so it is very strange that the build engine only has support for version 16 without specify a custom engine.

**Comment by bergmorten** [comment]
Still waiting for them to fix correct [environment name when deploying](https://github.com/aws-amplify/amplify-hosting/issues/3745). Seems that there are few bug fixs and updates on this UI page?

**Comment by josdejong** [comment]
> I still think Amplify should change default image to Linux 2023 (which I think has support for both node 18 and 20).

Definitely. Nodejs 16 is not a good default anymore, it's outdated and not maintained anymore 😅

**Comment by kachkaev** [comment]
Looks like Linux 2023 with Node 18 is available now (just not as a default): https://github.com/aws-amplify/amplify-hosting/issues/3773#issuecomment-1811288919

**Comment by bergmorten** [comment]
The new image 2023 worked great for me :-)

**Comment by josdejong** [comment]
Ah, yes, the `Amazon Linux:2023` image works now!
